.github/workflows/compile.yml

```diff
@@ -0,0 +1,76 @@
+name: Compile Arduino Blinky in all variants
+
+on:
+  push:
+    branches: '*'
+  pull_request:
+    branches: '*'
+
+jobs:
+  compile:
+    strategy:
+      fail-fast: false
+      matrix:
+        variant:
+          - nRF52DK
+          - BluzDK
+          - RedBearLab_nRF51822:version=1_0
+          - BBCmicrobit
+          - BBCmicrobitV2
+          - CalliopeMini
+          - Generic_nRF51822:chip=xxac
+          - Generic_nRF52832
+          - Generic_nRF52833
+          - OSHChip
+          - STCT_nRF52_minidev
+          - PCA1000X:board_variant=pca10000
+          - PCA1000X:board_variant=pca10001
+          - PCA1000X:board_variant=nrf6310
+          - nRF51Dongle:version=1_1_0
+          - Blend2
+          - BLENano
+          - BLENano2
+          - TinyBLE
+          - bluey
+          - hackaBLE
+          - hackaBLE_v2
+          - Sinobit
+          - DWM1001-DEV
+          - SeeedArchLink
+          - Beacon_PCA20006
+          - Waveshare_BLE400
+          - ng_beacon
+    runs-on: ubuntu-latest
+    name: Compile ${{ matrix.variant }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Create blinky sketch to compile
+        run: |
+          mkdir blinky
+          cat > blinky/blinky.ino << EOF
+            void setup() {
+              pinMode(LED_BUILTIN, OUTPUT);
+            }
+
+            void loop() {
+              digitalWrite(LED_BUILTIN, HIGH);
+              delay(1000);
+              digitalWrite(LED_BUILTIN, LOW);
+              delay(1000);
+            }
+          EOF
+      - uses: arduino/compile-sketches@v1
+        with:
+          cli-version: latest
+          verbose: true
+          enable-deltas-report: true
+          platforms: |
+            # First we have to install this core so that the tooling is installed
+            - name: "sandeepmistry:nRF5"
+              source-url: "https://sandeepmistry.github.io/arduino-nRF5/package_nRF5_boards_index.json"
+              version: latest
+            # Now, this will overwrite the core source to use the local version
+            - name: "sandeepmistry:nRF5"
+              source-path: .
```

Comment:
It's a bit funny having to first install a released version of the core, since we need to compile with the local code, so we have to then "reasign" the core to point to the local version.

It looks like this might be a limitation of the Arduino Action, since the official Arduino cores do the same:
- https://github.com/arduino/ArduinoCore-mbed/blob/3.1.1/.github/workflows/compile-examples.yml#L138-L143
- https://github.com/arduino/ArduinoCore-avr/blob/1.8.5/.github/workflows/compile-platform-examples.yml#L186-L191